src/store.ts

```diff
@@ -858,21 +827,35 @@ const createMlStore = (logging: Logging) => {
           },
         }),
         {
+          version: 1,
           name: "ml",
           partialize: ({
-            gestures: actions,
+            actions,
             project,
             projectEdited,
             settings,
             timestamp,
           }) => ({
-            gestures: actions,
+            actions,
             project,
             projectEdited,
             settings,
             timestamp,
             // The model itself is in IndexDB
           }),
+          migrate(persistedStateUnknown, version) {
+            if (version === 0) {
+              // Rename the "gestures" field to "action"
+              const persistedState = persistedStateUnknown as Store;
+              interface OldStore extends Omit<Store, "actions"> {
+                gestures?: ActionData[];
+              }
+              const prevPersistedState = persistedStateUnknown as OldStore;
+              persistedState.actions = prevPersistedState.gestures ?? [];
+              delete prevPersistedState.gestures;
```

Comment:
A slightly different way to write this so it's clear that we're always operating on the same object, but it's just as messy when it comes to type casting. Maybe there's a better options?

```suggestion
              interface OldStore extends Omit<Store, "actions"> {
                gestures?: ActionData[];
              }
              (persistedStateUnknown as Store).actions =
                (persistedStateUnknown as OldStore).gestures ?? [];
              delete (persistedStateUnknown as OldStore).gestures;
```

Reply:
I agree that your suggestion is much more readable. Maybe @microbit-matt-hillsdon might have suggestions?

Reply:
We tweaked this further after discussion. Resolving.